src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -0,0 +1,340 @@
+                    }
+                } catch (MatlabVersionNotFoundException e) {
+                    return FormValidation.warning(Message.getValue("Builder.invalid.matlab.root.warning"));
+                }
+            }
+            
+            
+            return FormValidation.ok();
+        };
+        
+        public FormValidation getFirstErrorOrWarning(
+                List<Function<String, FormValidation>> validations) {
+            if (validations == null || validations.isEmpty())
+                return FormValidation.ok();
+            try {
+                final String matlabRoot = Jenkins.getInstance()
+                        .getDescriptorByType(MatlabBuildWrapperDescriptor.class).getMatlabRootFolder();
+                for (Function<String, FormValidation> val : validations) {
+                    FormValidation validationResult = val.apply(matlabRoot);
+                    if (validationResult.kind.compareTo(Kind.ERROR) == 0
+                            || validationResult.kind.compareTo(Kind.WARNING) == 0) {
+                        return validationResult;
+                    }
+                }
+            }catch (Exception e) {
+                return FormValidation.warning(Message.getValue("Builder.invalid.matlab.root.warning"));
+            }
+           
+            return FormValidation.ok();
+        }
+    }
+
+    @Override
+    public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace,
+            @Nonnull Launcher launcher, @Nonnull TaskListener listener)
+            throws InterruptedException, IOException {
+        //Set the environment variable specific to the this build
+        setEnv(build.getEnvironment(listener));
+        
+        // Invoke MATLAB command and transfer output to standard
+        // Output Console
+
+        buildResult = execMatlabCommand(workspace, launcher, listener, getEnv());
+
+        if (buildResult != 0) {
+            build.setResult(Result.FAILURE);
+        }
+    }
+
+    private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher,
+            TaskListener listener, EnvVars envVars)
+            throws IOException, InterruptedException {
+    
+        ProcStarter matlabLauncher;
+        try {
+            // Get matlabroot set in wrapper class.
+            String matlabRoot = envVars.get("matlabroot");     
+            CommandConstructUtil cmdUtils = new CommandConstructUtil(launcher, matlabRoot);
+            matlabLauncher = launcher.launch().pwd(workspace).envs(envVars);
+            FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
+            if(launcher.isUnix()) {
+                matlabLauncher = launcher.launch().pwd(workspace).envs(envVars).cmds("/bin/bash","-c","./run_matlab_command.sh "+cmdUtils.constructCommandForTest(getInputArguments())).stdout(listener);
+                //Copy runner .sh for linux platform in workspace.

```

Comment:
I don't think we need to copy `run_matlab_command` to the workspace. I don't believe a user will need to be looking at this file. Can you use a temp folder instead?

Reply:
I can try using temp. but i guess it might need a round of discussion about where to create temp folder both for Linux & for Windows . also we need to provide runMatlabtest with an absolute path which would make the whole command look bit odd (not any of the concern though) . we could also explore the option of deleteing it from workspace after build completes.

Reply:
BATCH & SHELL uses 
**java.io.tmpdrive** system path to create a scratch file and they invoke it using the absolute path. as shown below 
![image](https://user-images.githubusercontent.com/47204011/75424071-fa0dd180-5965-11ea-8bc4-5519cf0a2d95.png)

Reply:
Hi Mark i have copied the runner file under tmp folder basically under system property "java.io.tmpdir"  and deleteing the file after build completion . this seems to be the common practice with batch and shell scripts files too. let me know if thinks this is ok.

Reply:
@mcafaro  the changes are in already. you may have your review comments.